sql/src/main/java/io/crate/expression/reference/file/LineContext.java

```diff
@@ -70,4 +79,23 @@ public void rawSource(byte[] bytes) {
         this.rawSource = bytes;
         this.parsedSource = null;
     }
+
+    public void currentUri(URI currentUri) {
+        this.currentUri = BytesRefs.toBytesRef(currentUri.toString());
+        // reset failure if new URI is processed
```

Comment:
Could we also explain why (instead of what)? Do we drop the failure then? Is it supposed to be recorded when a new URI is set?

Or maybe we should add a javadoc instead of an inline comment that this drops prior URI failures and that they've to be recorded separately